sklearn/inspection/_plot/partial_dependence.py

```diff
@@ -274,35 +300,47 @@ def convert_feature(fx):
                              'len(feature_names) = {0}, got {1}.'
                              .format(len(feature_names), i))
 
-    # compute averaged predictions
+    if ((isinstance(subsample, int) and subsample <= 0)
+        or (isinstance(subsample, float)
+            and (subsample <= 0 or subsample >= 1))):
```

Comment:
I think it would be more readable if we split the statement:


```python
if isinstance(subsample, numbers.Integral) and ....:
    raise ValueError('When an integer, ...')
elif isinstance(subsample, numbers.Real) and ...:
    raise ValueError('When a floating-point, ...')
```

Reply:
Done in https://github.com/scikit-learn/scikit-learn/pull/16619/commits/7e96ad9f59c68990ee628df4abba447509a6350d